### ISSUE TITLE
adding pagination_class to ModelViewSet is not returning paginated results

### ISSUE THREAD

**Issue by awwester** [tomchristie/django-rest-framework#4304]
I'm having issues with setting `pagination_class` on a single viewset based off of `ModelViewSet`. When I set the default pagination class in the settings, it works as expected. But I only want pagination on one viewset, and that is not working for me with a very simple example.

I'm surprised I haven't found anyone else that has reported this because I'm seeing it with a very simple example. Maybe I'm missing something or doing something wrong, but according to the [docs](http://www.django-rest-framework.org/api-guide/pagination/#setting-the-pagination-style) I should be able to just add the `pagination_class` property to any class based off of `GenericAPIView` and it should just work(tm).

## EDIT/UPDATE
After going through the source code this definitely seems like a bug to me. The problem is that `PageNumberPagination.get_page_size()` was returning `None`. At the top of the PageNumberPagination class is `page_size = api_settings.PAGE_SIZE` and for whatever reason, it's not getting overridden when defining on my ClipViewSet class.

I then set just the PAGE_SIZE in settings and then setting `pagination_class` on my class works.

```
# need to set PAGE_SIZE for paginating to work on viewset
REST_FRAMEWORK = {
    # 'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.PageNumberPagination',
    'PAGE_SIZE': '10'
}
```
If we can confirm this is a bug I'll get a PR going but need confirmation I'm not doing something dumb!

## Steps to reproduce
Create viewset based off of `ModelViewSet` using rest_framework 3.4.0 and django 1.9.7
Add `pagination_class = PageNumberPagination` to the class.

    class ClipViewSet(ModelViewSet):
        serializer_class = ClipSerializer
        queryset = Clip.objects.all()
        pagination_class = PageNumberPagination
        page_size = 10

## Expected behavior
The response should contain the paginated results

    "count": 542,
    "next": null,
    "previous": null,
    "results": [ ... ]

## Actual behavior
Returns response that is not paginated.

## Checklist
- [ ] I have verified that that issue exists against the `master` branch of Django REST framework.
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] This is not a usage question. (Those should be directed to the [discussion group](https://groups.google.com/forum/#!forum/django-rest-framework) instead.)
- [x] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](http://www.django-rest-framework.org/topics/third-party-resources/#about-third-party-packages) where possible.)
- [x] I have reduced the issue to the simplest possible case.
- [ ] I have included a failing test as a pull request. (If you are unable to do so we can still accept the issue.)

**Comment by tomchristie** [comment]
> I then set just the PAGE_SIZE in settings and then setting pagination_class on my class works.

That's the correct usage. You *do* need to set the `PAGE_SIZE` in the settings.

Your later example where you set a `page_size` view attribute is incorrect.

    class ClipViewSet(ModelViewSet):
        serializer_class = ClipSerializer
        queryset = Clip.objects.all()
        pagination_class = PageNumberPagination
        page_size = 10  # <--- Not valid or documented usage.

**Comment by awwester** [comment]
After looking into this after some sleep, I realize my initial thought was wrong. I needed to be making a custom pagination class and adding that pagination class that has the `page_size`. I made a mistake and thought `page_size` was to be added to the view.

However, that idea was sparked by the docs, and I do think there's a possible improvement to the docs.

In [Setting the Pagination Style](http://www.django-rest-framework.org/api-guide/pagination/#setting-the-pagination-style), it makes no reference that `PAGE_SIZE` is a required setting, and the second paragraph just doesn't work without getting into the source code and seeing why pagination isn't working after doing what the second paragraph says.

I think if we added a paragraph to that section referencing the need for the `PAGE_SIZE` it would be more clear. Just my 2 cents.
